python/cudf/cudf/pandas/_wrappers/pandas.py

```diff
@@ -1734,6 +1736,88 @@ def _unpickle_obj(pickled_args):
     return obj
 
 
+# Save the original __init__ methods
+_original_Series_init = cudf.Series.__init__
+_original_DataFrame_init = cudf.DataFrame.__init__
+_original_Index_init = cudf.Index.__init__
+
+
+def wrap_init(original_init):
+    @functools.wraps(original_init)
+    def wrapped_init(self, data=None, *args, **kwargs):
+        if is_proxy_object(data):
+            data = data.as_gpu_object()
+            if (
+                isinstance(data, type(self))
+                and len(args) == 0
+                and len(kwargs) == 0
+            ):
+                # This short-circuits the constructor to avoid
+                # unnecessary work when the data is already a
+                # proxy object of the same type.
+                # It is a common case in `cuml` and `xgboost`.
+                # For perf impact see:
+                # https://github.com/rapidsai/cudf/pull/17878/files#r1936469215
+                self.__dict__.update(data.__dict__)
+                return
+        original_init(self, data, *args, **kwargs)
+
+    return wrapped_init
+
+
+@functools.wraps(_original_DataFrame_init)
+def DataFrame_init_(self, data, index=None, columns=None, *args, **kwargs):
+    data_extracted = False
+    if is_proxy_object(data):
+        data = data.as_gpu_object()
+        data_extracted = True
```

Comment:
```suggestion
    data_is_proxy = is_proxy_object(data)
    if data_is_proxy:
        data = data.as_gpu_object()
```

Reply:
Could even walrus it but I don't like those in cases where the lifetimes are not obvious (i.e. we'd be relying on `data_is_proxy` persisting beyond the conditional where it is created, which is valid but less readable IMHO).